### PR TITLE
feat(onboarding-1.5): Invite modal updates

### PR DESCRIPTION
**Pull request by liyiy**
## Problem

1.5 Invite member modal updates 
- During ingestion user can now invite members and get the links without having to break out of the flow of the modal 
- Users can now add optional messages to the invite 

These changes only apply to onboarding 1.5 so the current invite modal is kept to the same existing functionality as much as possible

## Changes

ONBOARDING 1.5 FLOW 

- WITH EMAIL CONFIGURED

Screen Shot 2022-05-26 at 4 30 49 PM

- WITHOUT EMAIL CONFIGURED
Screen Shot 2022-05-26 at 5 38 57 PM

WITHOUT 1.5 FLOW

- WITH EMAIL CONFIGURED
Screen Shot 2022-05-26 at 5 30 37 PM

- WITHOUT EMAIL CONFIGURED
Screen Shot 2022-05-26 at 5 31 36 PM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/organization/Settings/InviteModal.tsx [hunk 1]
```diff
@@ -65,102 +71,257 @@ function InviteRow({ index, isDeletable }: { index: number; isDeletable: boolean
...
-                                Name <i>(optional)</i>
+                                {onboardingSidebarEnabled && !preflight?.email_service_available
+                                    ? 'Share link'
+                                    : 'Name (optional)'}
                             </b>
                         </Col>
                     </Row>
 
+                    {onboardingSidebarEnabled &&
+                        invites.map((invite: OrganizationInviteType) => {
```
Comment (liyiy) [inline frontend/src/scenes/organization/Settings/InviteModal.tsx:187]: These are the pending invites shown during ingestion so the user doesn't have to break out of the flow to view who's been invited. You cannot edit pending invites and can only delete/cancel them.

**Review by clarkus — DISMISSED** [review]
Still reviewing but one screen doesn't match the designs. Specifically the version where there is a banner at the bottom of the modal. That content should be in paragraph style below the header. It's in figma at https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=10499%3A43202

**Comment by liyiy** [comment]
> Still reviewing but one screen doesn't match the designs. Specifically the version where there is a banner at the bottom of the modal. That content should be in paragraph style below the header. It's in figma at https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=10499%3A43202

The banner has a link to set up email configuration, is that something we want to exclude from our new flow? There's also no expiration date warning for invites in the mocks, is that not useful to show users? 😮

**Comment by clarkus** [comment]
> The banner has a link to set up email configuration, is that something we want to exclude from our new flow?

No, but I do want to hear that as part of feedback during design rounds. This looks like the banner component taken from settings and placed at the bottom of the modal. The content is somewhat the same and it has the link, but the content from the designs is otherwise completely absent.

In the designs there are descriptions at the top of each modal that help set the purpose and clarify the constraints. That's the place to include this content. Happy to talk about this more, but this just doesn't match what was agreed to during the design rounds.

**Comment by liyiy** [comment]
> This looks like the banner component taken from settings and placed at the bottom of the modal. The content is somewhat the same and it has the link, but the content from the designs is otherwise completely absent.

The banner warning already existed in the invite modal I didn't add or remove anything to it

> In the designs there are descriptions at the top of each modal that help set the purpose and clarify the constraints. That's the place to include this content. Happy to talk about this more, but this just doesn't match what was agreed to during the design rounds.

ok

**Comment by clarkus** [comment]
> The banner warning already existed in the invite modal I didn't add or remove anything to it

OK. The same content is in the header. We can just append a link there. I'll update the design in figma just so it's accurate.

**Comment by clarkus** [comment]
I think I misunderstood the screenshots. Are the screenshots with the banner in the bottom what's out there now? Are we supporting that scenario going forward at all or are they completely replaced by these changes? I was thinking the feedback was about the configure email link but I think I misread this. 🙃

**Comment by liyiy** [comment]
@rcmarron 

Screen Shot 2022-05-27 at 4 22 00 PM

File: posthog/templates/email/invite.html [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
 </p>
+{% if invite.message %}
+<div class="invite-message">
+    <div class="invite-message-divider"></div>
+    {{ invite.message }}
+</div>
+{% endif %}
 <div class="mb mt text-center">
-    <a class="button" href="{{ invite }}">Join {{ invite.organization.name }} now</a>
+    <a class="button primary" href="{{ invite }}">Join now</a>
```
Comment (rcmarron) [inline posthog/templates/email/invite.html:16]: Is `{{ invite }}` right here? Would that give us the invite object instead of the link?

**Review by rcmarron — APPROVED** [review]

**Pull request merged**